avocado/core/test.py

```diff
@@ -1011,27 +1011,38 @@ def fetch_asset(self, name, asset_hash=None, algorithm=None,
         if expire is not None:
             expire = data_structures.time_to_seconds(str(expire))
 
+        # If name has no protocol or network locations, attempt to find
+        # the asset "by name" first. This is valid use case when the
+        # asset has been previously put into any of the cache
+        # directories, either manually or by the caching process
+        # itself.
```

Comment:
Ok, I see what you did here. :)

But this kind of usage that I was talking about: makes a significant impact on user's experience, IMO. Because, our current implementation implies that "fetch" is associated with "downloading an asset". For instance: depending on the runner debug.log could have messages saying to users that we are downloading the asset:

```
2021-02-05 21:19:15,554 asset            L0329 INFO | Asset not in cache, fetching it.
```
(Because of `plugins/assets.py`)

But I understood your valid points about my initial approach of having two different methods. So please, feel free to ignore my concern here, and let's try to get this merged and improve this on future versions.